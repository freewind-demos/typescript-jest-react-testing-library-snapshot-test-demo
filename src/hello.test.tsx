import React from 'react';
import {render, fireEvent} from "@testing-library/react";
import Hello from "./hello";


describe('Hello', () => {
  test('handle onChange', () => {
    const mockOnChange = jest.fn()
    const {container} = render(<Hello name='typescript' onChange={mockOnChange}/>)
    expect(container).toMatchSnapshot();
  });
})
