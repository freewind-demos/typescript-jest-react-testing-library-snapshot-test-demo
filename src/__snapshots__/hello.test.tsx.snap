// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello handle onChange 1`] = `
<div>
  <div>
    <div>
      Hello, 
      typescript
    </div>
    <input
      type="text"
      value="typescript"
    />
  </div>
</div>
`;
